app/src/ui/lib/button.tsx

```diff
@@ -139,12 +139,12 @@ export class Button extends React.Component<IButtonProps, {}> {
   }
 
   private onClick = (event: React.MouseEvent<HTMLButtonElement>) => {
-    if (this.props.type !== 'submit') {
-      event.preventDefault()
-    }
-
     if (this.props.onClick) {
       this.props.onClick(event)
     }
+
+    if (this.props.type !== 'submit') {
+      event.preventDefault()
+    }
```

Comment:
This change is necessary in order for the menu backed blankslate action to determine whether consumers of its `onClick` event wants to prevent launching the menu action.

I've gone through all the places in the app where we rely on the `defaultPrevented` property and none of them are related to the `Button` component so this feels like a safe change while at the same time following the conventions of other components.